### PR TITLE
add coinMP package

### PR DESCRIPTION
**Pull request by bgruening**

### CODE REVIEW

**Comment by bgruening** [comment]
@jakirkham Is it really needed to call it `CPL-1.0` to make the linter happy?

**Comment by bgruening** [comment]
Green, nice! Any merge appreciated I need this one for further downstream packages :)

**Comment by bgruening** [comment]
@jakirkham please! :)

File: recipes/coinmp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  build:
+    - toolchain
+    - pkgconfig
+    - readline
+    - zlib
+    - bzip2
+  run:
+    - readline
+    - zlib
+    - bzip2
```
Comment (jakirkham) [inline recipes/coinmp/meta.yaml]: So these need to be pinned according to this [list]( https://github.com/conda-forge/conda-forge.github.io/blob/af6e7b282efb2bf9b62fe10f8093d76b37c333e1/scripts/pin_the_slow_way.py#L29-L60 ). Please update in the feedstock and bump the build number.

File: recipes/coinmp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - pkgconfig
+    - readline
+    - zlib
+    - bzip2
```
Comment (jakirkham) [inline recipes/coinmp/meta.yaml]: Same as below.

File: recipes/coinmp/build.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+chmod +x ./CoinUtils/install-sh
+chmod +x ./Osi/install-sh
+chmod +x ./Clp/install-sh
+chmod +x ./Cgl/install-sh
+chmod +x ./Cbc/install-sh
+chmod +x ./CoinMP/install-sh
+chmod +x ./Data/Sample/install-sh
+
+make
+#make test
```
Comment (jakirkham) [inline recipes/coinmp/build.sh]: Any reason for this to be commented?

File: recipes/coinmp/build.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+sh ./configure --prefix=$PREFIX --enable-static --enable-gnu-packages
```
Comment (jakirkham) [inline recipes/coinmp/build.sh]: Does `--enable-gnu-packages` impact the licensing? If so, how?

File: recipes/coinmp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  run:
+    - readline
+    - zlib
+    - bzip2
+
+test:
+  commands:
+    - cbc ?
+    - clp ?
```
Comment (jakirkham) [inline recipes/coinmp/meta.yaml]: Are there headers and/or libraries we should be checking for too?

File: recipes/coinmp/meta.yaml (hunk 1 above)
Comment (bgruening) [inline recipes/coinmp/meta.yaml]: Why was smithy not complaining? Good to now thanks.

File: recipes/coinmp/build.sh (hunk 3 above)
Comment (bgruening) [inline recipes/coinmp/build.sh]: Some tests are failing and this is known upstream afaik.

File: recipes/coinmp/build.sh (hunk 4 above)
Comment (bgruening) [inline recipes/coinmp/build.sh]: Not that I know, is enables searching for GNU packages, not sure about the licensing in this case.

File: recipes/coinmp/meta.yaml (hunk 5 above)
Comment (bgruening) [inline recipes/coinmp/meta.yaml]: Do you have a nice example howto do so? Do you want a small program to be compiled?

**Pull request merged**